app/lib/features/events/pages/create_edit_event_page.dart

```diff
@@ -0,0 +1,532 @@
+import 'package:acter/common/providers/space_providers.dart';
+import 'package:acter/common/utils/routes.dart';
+import 'package:acter/common/utils/utils.dart';
+import 'package:acter/common/widgets/base_body_widget.dart';
+import 'package:acter/common/widgets/html_editor.dart';
+import 'package:acter/common/widgets/spaces/select_space_form_field.dart';
+import 'package:acter/features/events/providers/event_providers.dart';
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk.dart';
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:dart_date/dart_date.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::event::createOrEdit');
+
+class CreateEditEventPage extends ConsumerStatefulWidget {
+  final String? initialSelectedSpace;
+  final String? calendarId;
+
+  const CreateEditEventPage({
+    super.key,
+    this.initialSelectedSpace,
+    this.calendarId,
+  });
+
+  @override
+  ConsumerState<CreateEditEventPage> createState() =>
+      _CreateEditEventPageConsumerState();
+}
+
+class _CreateEditEventPageConsumerState
+    extends ConsumerState<CreateEditEventPage> {
+  final _eventFromKey = GlobalKey<FormState>();
+  final _eventNameController = TextEditingController();
+  final _startDateController = TextEditingController();
+  final _startTimeController = TextEditingController();
+  DateTime _selectedStartDate = DateTime.now();
+  TimeOfDay _selectedStartTime = TimeOfDay.now();
+  final _endDateController = TextEditingController();
+  final _endTimeController = TextEditingController();
+  DateTime _selectedEndDate = DateTime.now();
+  TimeOfDay _selectedEndTime = TimeOfDay.now();
+  EditorState textEditorState = EditorState.blank();
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((Duration duration) {
+      // if calendarId is not null that means Edit Event
```

Comment:
Nice, I am now thinking similar about having one page that deals with create edit functions of pins.

Reply:
Yes, one file for create and edit flow is nice but based on the below conditions.
- Code complexity doesn't gets increased with common create and edit flow.
- When there are chances of high amount of code duplication.